packages/utils/src/ast.ts

```diff
@@ -0,0 +1,274 @@
+      // There may be a better solution to avoid having to filter those out.
+      .filter((token) => token !== '' && token !== undefined)
+  );
+}
+
+/**
+ * Post process code with AST such that it can be evaluated in SES.
+ *
+ * Currently:
+ * - Makes all direct calls to eval indirect.
+ * - Handles certain Babel-related edge cases.
+ * - Removes the `Buffer` provided by Browserify.
+ * - Optionally removes comments.
+ * - Breaks up tokens that would otherwise result in SES errors, such as HTML
+ * comment tags `<!--` and `-->` and `import(n)` statements.
+ *
+ * @param ast - The AST to post process.
+ * @returns The modified AST.
+ */
+export function postProcessAST(ast: Node): Node {
+  const handler: TraverseOptions = {
+    enter(path) {
+      const { node } = path;
+
+      COMMENTS_KEYS.forEach((key) => {
+        const comments = node[key];
+        if (!comments) {
+          return;
+        }
+
+        // Break up tokens that could be parsed as HTML comment terminators. The
+        // regular expressions below are written strangely so as to avoid the
+        // appearance of such tokens in our source code. For reference:
+        // https://github.com/endojs/endo/blob/70cc86eb400655e922413b99c38818d7b2e79da0/packages/ses/error-codes/SES_HTML_COMMENT_REJECTED.md
+        comments.forEach((comment) => {
+          comment.value = comment.value
+            .replace(new RegExp(`<!${'--'}`, 'gu'), '< !--')
+            .replace(new RegExp(`${'--'}>`, 'gu'), '-- >')
+            .replace(/import(\(.*\))/gu, 'import\\$1');
+        });
+      });
+    },
+
+    FunctionExpression(path) {
+      const { node } = path;
+
+      // Browserify provides the `Buffer` global as an argument to modules that
+      // use it, but this does not work in SES. Since we pass in `Buffer` as an
+      // endowment, we can simply remove the argument.
+      //
+      // Note that this only removes `Buffer` from a wrapped function
+      // expression, e.g., `(function (Buffer) { ... })`. Regular functions
+      // are not affected.
+      //
+      // TODO: Since we're working on the AST level, we could check the scope
+      // of the function expression, and possibly prevent false positives?
+      if (node.type === 'FunctionExpression' && node.extra?.parenthesized) {
+        node.params = node.params.filter(
+          (param) => !(param.type === 'Identifier' && param.name === 'Buffer'),
+        );
+      }
+    },
+
+    CallExpression(path) {
+      const { node } = path;
+
+      // Replace `eval(foo)` with `(1, eval)(foo)`.
+      if (node.callee.type === 'Identifier' && node.callee.name === 'eval') {
+        path.replaceWith(
+          evalWrapper({
+            REF: node.callee,
+            ARGS: node.arguments,
+          }) as any,
+        );
+
+        // Need to skip the path here, to prevent the node from being visited
+        // again.
+        path.skip();
```

Comment:
This wouldn't visit the expressions inside the call and could potentially be an issue.